### PR TITLE
Fix for ldap3.core.exceptions.LDAPInvalidFilterError: malformed filter

### PR DESCRIPTION
**Pull request by vx-i**
Fix ldap3.core.exceptions.LDAPInvalidFilterError if LDAP filter string contains characters that needs to be escaped. For example:

displayName: Lastname, Firstname

master.cfg

# Active Directory user information
userInfoProvider = util.LdapUserInfo(
    uri='ldap://ldap.local:3268',
    bindUser='user@ldap.local',
    bindPw='pass',
    accountBase='OU=Companies,DC=ldap,DC=local',
    groupBase='OU=Companies,DC=ldap,DC=local',
    accountPattern='(&(objectClass=user)(sAMAccountName=%(username)s))',
    accountFullName='displayName',
    accountEmail='mail',
    groupMemberPattern='(&(objectClass=group)(member=%(dn)s))',
    groupName='Name',
    avatarPattern='(&(objectClass=user)(mail=%(email)s))',
    avatarData='thumbnailPhoto',
)

## Contributor Checklist:

* [x] I have updated the unit tests
* [x] I have created a file in the `master/buildbot/newsfragments` directory (and read the `README.txt` in that directory)
* [x] I have updated the appropriate documentation (not applicable)

### CODE REVIEW

**Review by p12tic — CHANGES_REQUESTED** [review]
Tests need to be updated.

Looks good otherwise.

**Comment by vx-i** [comment]
Good evening all, I think I need some help with tests. Because ldap3 is faked within tests, I get this error:

builtins.AttributeError: module 'ldap3' has no attribute 'utils'

How can I get rid of this?

**Comment by p12tic** [comment]
@vx-i Indeed. Looking into history it seems that the reason for this weird whole-module mocking in tests was that in the past ldap library was C-based, which I assume would have resulted in a hard dependency of that library in tests.

ldap3 is pure-python library, so this is no longer needed.

I think you could just mock get_config_parameter using the usual mock.patch in the test and remove the current whole module patching. Then the tests should work.

Thanks!

**Comment by vx-i** [comment]
@p12tic The tests still failing cause ldap3 is not in the requirements-ci.txt. Is there a reason why it is not in the list?

**Comment by p12tic** [comment]
@vx-i You're right, could you add it to requirements-ci.txt? Thanks.

File: master/buildbot/test/unit/www/test_ldapuserinfo.py [hunk 1]
```diff
@@ -91,7 +89,8 @@ def makeUserInfoProvider(self):
         self.userInfoProvider = ldapuserinfo.LdapUserInfo(
             uri="ldap://uri", bindUser="user", bindPw="pass",
             accountBase="accbase", groupBase="groupbase",
-            accountPattern="accpattern", groupMemberPattern="groupMemberPattern",
+            accountPattern="accpattern",
```
Comment (p12tic) [inline master/buildbot/test/unit/www/test_ldapuserinfo.py:92]: I think this wasn't needed for the fix you're doing was it? Buildbot uses 100 column line length limit so the current code was fine. Could you revert this change here and in a couple of places below?

**Review by p12tic — CHANGES_REQUESTED** [review]
Just a single nitpick, but otherwise looks great. Thank you!

File: master/buildbot/test/unit/www/test_ldapuserinfo.py (hunk 1 above)
Comment (vx-i) [reply master/buildbot/test/unit/www/test_ldapuserinfo.py]: Sorry, I didn't know. Fixed.

**Review by p12tic — APPROVED** [review]

**Comment by p12tic** [comment]
I've rebased on top of newest master and fixed a couple of nitpicks that I still had.
